### PR TITLE
Use named defines instead of hard coded numbers.

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

**Comment by Dead2** [comment]
`MAX_BITS` and `MAX_WBITS` are pretty similar looking. Should we change naming to something with a more clear distinction, or is it not needed? Personally I had to do a double-take when reviewing inflate.c, as I misread at first.
`MAX_BITS_WIN` would be more clear, but unfortunate since that would actually mean renaming an existing define.
In any case, this is not a deal-breaker for me, but something I think should be considered before merging.

Other than that I love it, I always hated magic numbers that might or might not mean what you think they do.

**Comment by nmoinvaz** [comment]
Yeah they are very similar; I think the names are okay. I didn't want to deviate too much from madler/zlib named defines. This PR is in preparation for PR #1333.

File: inffast_tpl.h [hunk 1]
```diff
@@ -184,12 +184,13 @@ void Z_INTERNAL INFLATE_FAST(PREFIX3(stream) *strm, uint32_t start) {
...
             if (op & 16) {                      /* distance base */
                 dist = here->val;
-                op &= 15;                       /* number of extra bits */
+                op &= MAX_BITS;                 /* number of extra bits */
                 if (bits < op) {
                     hold |= load_64_bits(in, bits);
                     in += 6;
                     bits += 48;
                 }
+                op &= MAX_BITS;                 /* number of extra bits */
```
Comment (mtl1979) [inline inffast_tpl.h]: There is no explanation why this line was added, and we generally discourage mixing cleanup and code changes.

**Comment by mtl1979** [comment]
I'm generally in favor of merging defines that are used for same purpose in different files and moving them to place where the preprocessor define can be included from in all needed files. I've already said I'm not big fan/supporter of magic numbers and prefer preprocessor defines as those can be documented better and improve readability and reduce confusion if same magic number is used for different purposes.

File: inffast_tpl.h (hunk 1 above)
Comment (nmoinvaz) [reply inffast_tpl.h]: Good catch, thank you. Fixed.

**Review by Dead2 — APPROVED** [review]

**Pull request merged**